html-aam/roles-contextual.html

```diff
@@ -0,0 +1,54 @@
+<!doctype html>
+<html>
+<head>
+  <title>HTML-AAM Contextual-Specific Role Verification Tests</title>
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <script src="/resources/testdriver.js"></script>
+  <script src="/resources/testdriver-vendor.js"></script>
+  <script src="/resources/testdriver-actions.js"></script>
+  <script src="/wai-aria/scripts/aria-utils.js"></script>
+</head>
+<body>
+
+
+<p>Tests contextual computedrole mappings defined in <a href="https://w3c.github.io/html-aam/">HTML-AAM</a>, where the returned computed role is expected to change based on the context. Most test names correspond to a unique ID defined in the spec.<p>
+
+<p>These should remain in alphabetical order.</code></p>
+
+
```

Comment:
Adding some notes to ease reviewers' understanding of the PR. 

All of the tests in this new file were preexisting but moved from `./html-aam/roles.html`

Reply:
This new file `./html-aam/roles-contextual.html` is limited to testing HTML tag names whose computed role is expected to differ when in different contexts... such as `<a>` with or without an `href` attribute.

Reply:
One exception is that "orphaned" roles (e.g. `<li>` outside a list) are not tested there, since there are so many potential orphan scenarios that are more relevant to include alongside their respective non-orphaned tests.